plugins/CorePluginsAdmin/Controller.php

```diff
@@ -251,7 +251,11 @@ private function createPluginsOrThemesView($template, $themesOnly)
 
     public function plugins()
     {
+        $marketplace = StaticContainer::get('Piwik\Plugins\Marketplace\Marketplace');
         $view = $this->createPluginsOrThemesView('plugins', $themesOnly = false);
+        $view->isInstallAllPaidPluginsVisible = $marketplace->isInstallAllPaidPluginsVisible();
+        $view->paidPluginsToInstallAtOnce = $marketplace->getPaidPluginsToInstallAtOnce();
+        $view->installAllPluginsNonce = Nonce::getNonce(\Piwik\Plugins\Marketplace\Controller::INSTALL_NONCE);
```

Comment:
I'm wondering if it might be smarter to change the InstallButton component to fetch all required information through a ajax request to a controller action (NOT api, due to the required token) instead of passing that through all controllers, views and vue components. Would be an additional request, but imho a lot simpler to reuse everywhere.

Reply:
@sgiehl Updated as suggested.